.github/workflows/test.yml

```
@@ -0,0 +1,87 @@
+name: Quality Assurance
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-linux:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        platform: [x32, x64]
+        compiler: [gcc, clang]
+    steps:
+      - name: Setup Dependencies
+        run: |
+          sudo apt-get update -y -qq
+          sudo apt-get install -y gcc g++ make autoconf automake make libyajl-dev libxml2-dev libmaxminddb-dev libpcre2-dev libpcre2-8-0 libpcre2-16-0 libpcre2-32-0 libcurl4-gnutls-dev jq wget
+      - name: Install ModSecurity library
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          modsecver=`gh release view -R owasp-modsecurity/ModSecurity -q .tagName --json tagName`
+          gh release download -p "*.tar.gz" -R owasp-modsecurity/ModSecurity $modsecver -O - | tar -xzf -
+          cd modsecurity-$modsecver
+          ./configure --disable-lmdb --prefix=/usr
+          make -j $(nproc)
+          sudo make install
+          cd
+      - uses: actions/checkout@v2
```

If you don't want to use hashes, that's your decision, but at least use the latest version of the app:
```suggestion
      - uses: actions/checkout@v4
```